consensus/e2e_tests/utils_test.go

```diff
@@ -673,68 +700,78 @@ func WaitForNodeToSync(
 	unsyncedNode *shared.Node,
 	allNodes IdToNodeMapping,
 	targetHeight uint64,
-) error {
+) {
 	currentHeight := unsyncedNode.GetBus().GetConsensusModule().CurrentHeight()
 
-	for i := currentHeight; i <= targetHeight; i++ {
-		blockRequest, err := waitForNodeToRequestMissingBlock(t, clck, eventsChannel, allNodes, currentHeight, targetHeight)
-		if err != nil {
-			return err
-		}
+	for currentHeight < targetHeight {
+		// waiting for unsynced node to request missing block
+		blockRequests, err := waitForNodeToRequestMissingBlock(t, clck, eventsChannel)
+		require.NoError(t, err)
 
-		blockResponse, err := waitForNodeToReceiveMissingBlock(t, clck, eventsChannel, allNodes, blockRequest)
-		if err != nil {
-			return err
-		}
+		// broadcast one of the requests to all nodes
+		P2PBroadcast(t, allNodes, blockRequests[0])
```

Comment:
Why are we only broadcasting `[0]`? DItto below

- Should we be looping over all of them and sending each one?
- Should we assert there's only only response if we're only using the 0th one?